presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -77,29 +140,92 @@ protected Void visitLogicalBinaryExpression(LogicalBinaryExpression node, Materi
         return super.visitLogicalBinaryExpression(node, context);
     }
 
-    protected static final class MaterializedViewPlanValidatorContext
+    @Override
+    protected Void visitComparisonExpression(ComparisonExpression node, MaterializedViewPlanValidatorContext context)
+    {
+        if (!context.isProcessingJoinNode()) {
+            return super.visitComparisonExpression(node, context);
+        }
+
+        if (!node.getOperator().equals(ComparisonExpression.Operator.EQUAL)) {
+            throw new SemanticException(NOT_SUPPORTED, node, "Only EQUAL join is supported for materialized view.");
+        }
+
+        if (isValidationOnly) {
+            return super.visitComparisonExpression(node, context);
+        }
+
+        Field left = analysis.get().getScope(context.getTopJoinNode()).tryResolveField(node.getLeft())
```

Comment:
same as set operation, lets keep the analysis to populate base column mapping in a separate check/method. that allows us to have only one place for             return super.visitComparisonExpression(node, context);

Reply:
Revised and put the base column mapping population part into a separate method, so that there's only one call to super.visitComparisonExpression(node, context). Let me know if you meant differently.